File: client/java/src/avt/PopupName.java
package avt;

import javax.microedition.lcdui.Graphics;
import main.Canvas;

public final class PopupName extends SubObject {
   private String name;
   private byte num;
   private byte c = 0;

   public PopupName(String var1, int var2, int var3) {
      super.catagory = 8;
      super.x = var2;
      super.y = var3;
      this.name = var1;
      this.num = (byte)CRes.rnd(8);
   }

   public final void update() {
      ++this.num;
      if (this.num >= 8) {
         this.num = 0;
      }

   }

   public final void paint(Graphics var1) {
      if (OptionScr.gI().mapFocus[1] != 1 && Canvas.welcome == null && super.x * MyObject.hd >= AvCamera.gI().xCam && super.x * MyObject.hd <= AvCamera.gI().xCam + Canvas.w && super.y * MyObject.hd >= AvCamera.gI().yCam && super.y * MyObject.hd <= AvCamera.gI().yCam + Canvas.h + 10 && Canvas.currentMyScreen != MainMenu.gI()) {
         var1.drawImage(LoadMap.imgShadow, super.x * MyObject.hd, super.y * MyObject.hd, 3);
         if (MiniMap.gI().imgArrow != null) {
            int var10002 = super.x * MyObject.hd;
            int var10003 = (super.y - 10 + this.num / 2) * MyObject.hd;
            MiniMap.gI().imgArrow.drawFrame(0, var10002, var10003, 0, 33, var1);
         }

         Canvas.smallFontYellow.drawString(var1, this.name, super.x * MyObject.hd, (super.y - 32 + this.num / 2) * MyObject.hd, 2);
      }

   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/PopupName.cs
public class PopupName : SubObject
{
	private string name;

	public sbyte num;

	public sbyte iPrivate;

	public PopupName(string name, int x, int y)
	{
		base.x = x;
		base.y = y;
		this.name = name;
		num = (sbyte)CRes.rnd(8);
	}

	public override void update()
	{
		num++;
		if (num >= 8)
		{
			num = 0;
		}
	}

	public override void paint(MyGraphics g)
	{
		if (OptionScr.gI().mapFocus[1] != 1 && (!MainMenu.gI().isGO || iPrivate != 0) && !((float)(x * MyObject.hd) < AvCamera.gI().xCam) && !((float)(x * MyObject.hd) > AvCamera.gI().xCam + (float)Canvas.w) && !((float)(y * MyObject.hd) < AvCamera.gI().yCam) && !((float)(y * MyObject.hd) > AvCamera.gI().yCam + (float)Canvas.hCan + 10f))
		{
			g.drawImage(LoadMap.imgShadow, x * MyObject.hd, y * MyObject.hd, 3);
			if (MiniMap.imgArrow != null)
			{
				MiniMap.imgArrow.drawFrame(0, x * MyObject.hd, (y - 10 + num / 2) * MyObject.hd, 0, MyGraphics.BOTTOM | MyGraphics.HCENTER, g);
			}
			Canvas.smallFontYellow.drawString(g, name, x * MyObject.hd, (y - 32 + num / 2) * MyObject.hd, 2);
		}
	}
}
